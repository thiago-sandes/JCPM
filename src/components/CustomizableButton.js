import React from 'react';
import {View, Button} from 'react-native';

const CustomizableButton = (props) => {
  const {style, onPress} = props;
  return (
    <View style={style}>
      <Button
        title="Voltar"
        onPress={() => {
          onPress();
        }}
      />
    </View>
  );
};

export default CustomizableButton;
