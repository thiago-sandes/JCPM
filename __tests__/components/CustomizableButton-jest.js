import React from 'react';
import {Button} from 'react-native';
import renderer from 'react-test-renderer';

// Components
import CustomizableButton from '../../src/components/CustomizableButton';

// Mocks
const customizableButtonMock = {
  style: {
    marginVertical: '2%',
    alignSelf: 'center',
  },
  onPress: jest.fn(),
};

describe('CustomizableButton component tests', () => {
  test('CustomizableButton snapshot', () => {
    const component = renderer.create(
      <CustomizableButton
        onPress={customizableButtonMock.onPress}
        style={customizableButtonMock.style}
      />,
    );

    expect(component.toJSON()).toMatchSnapshot();
  });

  test('CustomizableButton onPress', () => {
    const component = renderer.create(
      <CustomizableButton
        onPress={customizableButtonMock.onPress}
        style={customizableButtonMock.style}
      />,
    );

    component.root.findByType(Button).props.onPress();
    expect(customizableButtonMock.onPress).toHaveBeenCalled();
  });
});
