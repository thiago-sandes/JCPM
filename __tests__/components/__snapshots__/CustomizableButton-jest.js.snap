// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomizableButton component tests CustomizableButton snapshot 1`] = `
<View
  style={
    Object {
      "alignSelf": "center",
      "marginVertical": "2%",
    }
  }
>
  <View
    accessibilityRole="button"
    accessibilityState={Object {}}
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Array [
          Object {},
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#007AFF",
              "fontSize": 18,
              "margin": 8,
              "textAlign": "center",
            },
          ]
        }
      >
        Voltar
      </Text>
    </View>
  </View>
</View>
`;
